dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/redshift/RedshiftRawLogsConnector.java

```diff
@@ -97,14 +99,21 @@ public void addTasksTo(List<? super Task<?>> out, ConnectorArguments arguments)
         // 1. STL_QUERY has starttime, queryid, but text is 4000 char wich is useless
         // 2. STL_QUERY_TEXT has xid+squence+text which reconstructs query, but no starttime.
         // STL_QUERY is 1 row per query ; SQL_QUERY_TEXT is multi rows per query, using sequence and xid
-        String queryTemplateQuery
-                = "SELECT userid, xid, pid, query, trim(label) as label, starttime, endtime, sequence, text"
-                + " FROM STL_QUERY join STL_QUERYTEXT using (userid, xid, pid, query) WHERE ##";
+        List<String> queryTemplateColumns = new ArrayList<>(ImmutableList.<String>of(
+            "userid", "xid", "pid", "query", "trim(label) as label", "starttime", "endtime",
+            "sequence", "text"));
+        List<String> queryTemplateOrderBy = new ArrayList<>();
 
         if (arguments.isAssessment()) {
-            queryTemplateQuery += " ORDER BY starttime, query, sequence";
+            queryTemplateColumns.add("aborted");
+            queryTemplateOrderBy.addAll(ImmutableList.of("starttime", "query", "sequence"));
         }
 
+        String queryTemplateQuery = String.format(
+            "SELECT %s FROM STL_QUERY join STL_QUERYTEXT using (userid, xid, pid, query) WHERE ##%s",
+            Joiner.on(", ").join(queryTemplateColumns),
+            queryTemplateOrderBy.isEmpty() ? "" : (" ORDER BY " + Joiner.on(", ").join(queryTemplateOrderBy)));
```

Comment:
Truth be told, why not just have two static strings, much simpler that way, much easier to debug, too.

Reply:
I disagree because it would then require to keep them in sync and it would be easier for the parts that should be the same to be only changed in one of them. I'm not a big fan of constructing SQL statements out of strings. IMHO we should have a SQL builder, but that would be a bigger change. So, I'm just mimicking the important parts of that here.

I'd like to rather have something like this.

```java
SqlBuilder.select("STL_QUERY")
  .columns("userid", "xid", "pid")
  .columnAs("trim(label)", "trim")
  .orderBy("starttime")
```

Reply:
True enough, but we don't have that gadget, nor any interest in building it and maintaining it :-)